pkg/storage/couchdb/couchdbstore_test.go

```diff
@@ -290,6 +297,110 @@ func TestCouchDBStore(t *testing.T) {
 		itr = store.Iterator("abc_", "mno_123")
 		verifyItr(t, itr, 6, "")
 	})
+
+	t.Run("Test CouchDB store query", func(t *testing.T) {
+		t.Run("Successfully query using index", func(t *testing.T) {
+			prov, err := NewProvider(couchDBURL)
+			require.NoError(t, err)
+			store, err := prov.OpenStore(randomKey())
+			require.NoError(t, err)
+
+			couchDBStore, ok := store.(*CouchDBStore)
+			require.True(t, ok, "failed to assert store as a CouchDBStore")
+
+			testJSONPayload := []byte(`{"employeeID":1234,"name":"Mr. Aries"}`)
+
+			err = store.Put("sampleDBKey", testJSONPayload)
+			require.NoError(t, err)
+
+			const testDesignDoc = "TestDesignDoc"
+			const testIndexName = "TestIndex"
+			const fieldToIndex = "payload.employeeID"
+			err = couchDBStore.db.CreateIndex(context.Background(), testDesignDoc, testIndexName,
+				`{"fields": ["`+fieldToIndex+`"]}`)
+			require.NoError(t, err)
+
+			itr, err := store.Query(`{
+		   "selector": {
+		       "` + fieldToIndex + `": 1234
+		   },
+			"use_index": ["` + testDesignDoc + `", "` + testIndexName + `"]
+		}`)
+			require.NoError(t, err)
+
+			ok = itr.Next()
+			require.True(t, ok)
+
+			value := itr.Value()
+			require.NoError(t, err)
+			require.Equal(t, testJSONPayload, value)
+
+			ok = itr.Next()
+			require.NoError(t, err)
+			require.False(t, ok)
+
+			itr.Release()
+			require.NoError(t, err)
+		})
+		t.Run("Successful query, but the specified index isn't valid for the query", func(t *testing.T) {
+			prov, err := NewProvider(couchDBURL)
+			require.NoError(t, err)
+			store, err := prov.OpenStore(randomKey())
+			require.NoError(t, err)
+
+			couchDBStore, ok := store.(*CouchDBStore)
+			require.True(t, ok, "failed to assert store as a CouchDBStore")
+
+			testJSONPayload := []byte(`{"employeeID":1234,"name":"Mr. Aries"}`)
+
+			err = store.Put("sampleDBKey", testJSONPayload)
+			require.NoError(t, err)
+
+			const testDesignDoc = "TestDesignDoc"
+			const testIndexName = "TestIndex"
+			const fieldToIndex = "name"
+			err = couchDBStore.db.CreateIndex(context.Background(), testDesignDoc, testIndexName,
+				`{"fields": ["`+fieldToIndex+`"]}`)
+			require.NoError(t, err)
+
+			itr, err := store.Query(`{
+		   "selector": {
+		       "payload.employeeID": 1234
+		   },
+			"use_index": ["` + testDesignDoc + `", "` + testIndexName + `"]
+		}`)
+			require.NoError(t, err)
+
+			ok = itr.Next()
+			require.True(t, ok)
+
+			value := itr.Value()
+			require.NoError(t, err)
```

Comment:
Which err are you checking here? There are few references to same err bellow ...

Reply:
Oops. Good catch - those weren't supposed to be there. Fixed!